src/app/campaigns/campaign-edit/campaign-edit.component.ts

```diff
@@ -62,6 +65,11 @@ export class CampaignEditComponent {
             documents: asset.value.documents,
             newDocuments: [],
           })
+          this.newsUrls.clear()
+          this.newsUrls.markAsPristine()
+          asset.value.newsURLs.forEach((value) => this.newsUrls.push(this.fb.group({
+            url: [value, Validators.required]
+          })))
```

Comment:
can this be added in above `setValue` method? just asking, I've never experiment with formarrays.

Reply:
I tried it like that and it didn't work... I think it complained that it has no form controls defined in the array, so I figured they need to be created.